### ISSUE TITLE
getSourceExts() breaks typical module resolution

### ISSUE THREAD

**Issue by hypest** [kristerkari/react-native-sass-transformer#5]
SCSS file gets loaded instead of the intended JS file.

#### Context
When exporting an `getSourceExts()` [as per the README.md](https://github.com/kristerkari/react-native-sass-transformer/blob/master/README.md#step-2-configure-the-react-native-packager), the SCSS extensions take precedence before the normal `js`, `json` file extensions Metro looks up. [See this relevant commit on the RN project](https://github.com/facebook/react-native/commit/33a893e18b9cc59bbfaf86b3f0089fe49913b5b8).

When importing a local module without specifying an extension (perfectly valid use case as to allow for the platform specific file to get loaded) the bundler resolves the SCSS file first, the transformer produces JS code but the intended JS module is never imported. No error is raised until later, when some code tries to execute code from the "missed" module.

**Comment by hypest** [comment]
As a rather hacky workaround, one can define the `getSourceExts()` function as such:

```js
module.exports = {
  getTransformModulePath() {
    return require.resolve("react-native-sass-transformer");
  },
  getSourceExts() {
    return ["js", "json", "scss", "sass"];
  }
};
```

so the JS and JSON extensions still take precedence over the SCSS ones.

I wonder if there's a cleaner way to solve this issue. Ideas?

**Comment by kristerkari** [comment]
@hypest Thanks for this! I would really recommend to always specify the filetype when importing files. 

I think that your workaround might be as good as it gets if you don't specify the file extension when importing. 

I've been using the file extensions in all my projects and with the help of [babel-plugin-react-native-platform-specific-extensions](https://github.com/kristerkari/babel-plugin-react-native-platform-specific-extensions) I have been also using platform specific extensions.

**Comment by kristerkari** [comment]
I guess that for now I'll just update the docs with your workaround, and let's see if it's possible to do a better fix for the issue.

It sucks that React Native gives such limited options for specifying options for Metro.

**Comment by kristerkari** [comment]
I'm definitely open for any new features that can be added to the sass transformer. 👍 

So far what has been holding me back has been lack of time and the fact that I haven't found a good way to pass options to `node-sass`. The good news are that `node-sass` is open for adding a project config file (https://github.com/sass/node-sass/issues/1144#issuecomment-372022394), which would allow the options to be defined outside of the transformer. 

Just ping me if you want to discuss any specific features or if you have something ready that you want to contribute.

**Comment by hypest** [comment]
> I guess that for now I'll just update the docs with your workaround, and let's see if it's possible to do a better fix for the issue.

Oh, thanks! Btw, let me know if there's a cleaner way to rewrite this issue's description as it apparently wasn't super clear.

> It sucks that React Native gives such limited options for specifying options for Metro.

True. We find ourselves resorting to Babel plugins to customize behavior which might be more suitable for Metro, the bundler.

**Comment by kristerkari** [comment]
This is now properly solved since React Native 0.57+ where you have to add your extensions to the existing ones:

```js
sourceExts: [...sourceExts, "scss", "sass"]
```
